src/D2L.CodeStyle.Analyzers/Immutability/MutabilityInspector.cs

```diff
@@ -517,19 +523,38 @@ private static T GetDeclarationSyntax<T>( ISymbol symbol )
 			return decl;
 		}
 
-		private MutabilityInspectionResult InspectMemberRecursive(
+		private MutabilityInspectionResult InspectMember(
 			ISymbol symbol,
 			HashSet<ITypeSymbol> typeStack
 		) {
-			// if the member is audited or unaudited, ignore it
+			// if the member is audited or unaudited, check if valid
 			if( Attributes.Mutability.Audited.IsDefined( symbol ) ) {
+
+				var auditedInspectionResult = InspectMemberRecursive( symbol, typeStack );
+
+				if( !auditedInspectionResult.IsMutable ) {
+					var auditedAttributeSyntax = Attributes.Mutability.Audited.GetAttributeSyntax( symbol );
+
+					return MutabilityInspectionResult
+						.NotMutable()
+						.WithUnnecessaryAnnotations( ImmutableHashSet.Create( auditedAttributeSyntax ) );

```

Comment:
Think I'm going to update this so I can say something like,
`MutabilityInspectionResult.Audited( inspectionResult, auditedAttributeSyntax )`